### PR TITLE
u1986f code review 2

### PR DESCRIPTION
**Pull request by PerryBaran**

### CODE REVIEW

File: test/kata4.humanCatDogYears.test.js [hunk 1]
```diff
@@ -1,3 +1,8 @@
 const { humanCatDogYears } = require("../src");
 
-// Look Ma, no handlebars!!!
+describe("humanCatDogYears", () => {
+  it("returns Array of human, cat and dog years when passed human years", () => {
+    expect(humanCatDogYears(12)).toEqual([12, 64, 74]);
+    expect(humanCatDogYears(17)).toEqual([17, 84, 99]);
+  });
+});
```
Comment (PerryBaran) [inline test/kata4.humanCatDogYears.test.js:8]: you should add more tests for edge cases, e.g. when human years is 0, 1 or 2. e.g.
```
it("returns correct years when human years is equal to 0", () => {
  expect(humanCatDogYears(0)).toEqual([0, 0 ,0]);
}

it("returns correct years when human years is equal to 1", () => {
  expect(humanCatDogYears(1)).toEqual([1, 15, 15]);
}

it("returns correct years when human years is equal to 2", () => {
  expect(humanCatDogYears(1)).toEqual([2, 24, 24]);
}
```

File: src/kata4.humanCatDogYears.js [hunk 2]
```diff
@@ -1,3 +1,13 @@
...
 
-module.exports = humanCatDogYears;
+  const animalYears = function (catOrDogMultiplier) {
+    return catOrDogMultiplier * (number - 2) + firstTwoYears;
+  };
+
+  const catYears = animalYears(4);
+  const dogYears = animalYears(5);
+  return Array.of(number, catYears, dogYears);
+};
```
Comment (PerryBaran) [inline src/kata4.humanCatDogYears.js:11]: this works if human years is equal to 2 or more. However, you get the incorrect output if human years is 0 or 1.

File: src/kata4.humanCatDogYears.js [hunk 3]
```diff
@@ -1,3 +1,13 @@
-const humanCatDogYears = (number) => {};
+const humanCatDogYears = (number) => {
+  const firstTwoYears = 15 + 9;
 
-module.exports = humanCatDogYears;
+  const animalYears = function (catOrDogMultiplier) {
+    return catOrDogMultiplier * (number - 2) + firstTwoYears;
+  };
```
Comment (PerryBaran) [inline src/kata4.humanCatDogYears.js:6]: I like that you've used a helper function to reduce repetition

File: src/kata4.humanCatDogYears.js [hunk 4]
```diff
@@ -1,3 +1,13 @@
...
+
+  const catYears = animalYears(4);
+  const dogYears = animalYears(5);
+  return Array.of(number, catYears, dogYears);
```
Comment (PerryBaran) [inline src/kata4.humanCatDogYears.js:10]: instead of using `Array.of(...)`, it would be simpler to just write 
```
return [number, catYears, dogYears];
```

File: src/kata6.joinNames.js [hunk 5]
```diff
@@ -1,3 +1,13 @@
-const joinNames = (namesObj) => {};
+const joinNames = (namesObj) => {
+  let string = "";
+  for (let i = 0; i < namesObj.length; i++) {
+    if (i === namesObj.length - 2) {
+      string += `${namesObj[i].name} & ${namesObj[i + 1].name}`;
+      break;
+    }
+    string += `${namesObj[i].name}, `;
+  }
```
Comment (PerryBaran) [inline src/kata6.joinNames.js:9]: instead of using a `break` here, you could instead change it to an `if else` and change your condition to `i < namesObj.length - 1`, e.g.
```
  for (let i = 0; i < namesObj.length - 1; i++) {
    if (i === namesObj.length - 2) {
      string += `${namesObj[i].name} & ${namesObj[i + 1].name}`;
    } else {
      string += `${namesObj[i].name}, `;
    }
  }
```

File: test/kata6.joinNames.test.js [hunk 6]
```diff
@@ -1,5 +1,19 @@
...
+        },
+        {
+          name: "Ron",
+        },
+        {
+          name: "Hermione",
+        },
+      ])
+    ).toBe("Harry, Ron & Hermione");
+  });
```
Comment (PerryBaran) [inline test/kata6.joinNames.test.js:18]: I would add more tests for edge changes e.g. if the namesObj length was 1, 2 or more than 3. Currently, you return the wrong result if namesObj length is 1.

Although your code passes when namesObj length is 2 or more than 3, you should still write tests for those edge cases just to be certain and to catch an error if you refactor the function later.

File: src/kata6.joinNames.js [hunk 7]
```diff
@@ -1,3 +1,13 @@
...
+  return string;
+};
```
Comment (PerryBaran) [inline src/kata6.joinNames.js:11]: the simplest way to fix your code for if the namesObj has a length of 1 is to add the line `if (namesObj.length === 1) return namesObj[0].name;` at the top of the function, before `let string...`

File: src/kata4.humanCatDogYears.js (hunk 2 above)
Comment (PerryBaran) [reply src/kata4.humanCatDogYears.js:11]: the simplest way to fix this it just to add 2 `if` statements at the top of your function, before `const firstTwoYears = ...`, e.g.
```
if (number === 0) return [0, 0, 0];
if (number === 1) return [1, 15, 15];
```

**Review by PerryBaran — COMMENTED** [review]
Good Stuff though you've overlooked some edge cases. I've left suggestions on more tests you could add to test for the edges cases and suggestions on how to quickly fix your problems.

One other thing that I haven't mentioned elsewhere is that there are a lot of linting (formatting) issues with your project. If you have red squiggly lines in your code and/or the file name shows up as red in your explorer you need to fix these issues. If you hover your mouse over the red squiggly line a popup will appear that will give you more information about the problem. Most of the time there is a button at the bottom of this popup with the text `Quick Fix... (Ctrl+.)` that, if clicked, will usually fix your problem for you.
